07-Reverse_Integer.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+'''
+Given a 32-bit signed integer, reverse digits of an integer.
+
+Example 1:
+
+Input: 123
+Output: 321
+Example 2:
+
+Input: -123
+Output: -321
+Example 3:
+
+Input: 120
+Output: 21
+
+Note:
+Assume we are dealing with an environment which could only store integers within
+the 32-bit signed integer range: [−231,  231 − 1]. For the purpose of this problem,
+assume that your function returns 0 when the reversed integer overflows.
+'''
+
+class Solution:
+    def reverse(self, x):
+        """
+        :type x: int
+        :rtype: int
+        """
+        reversedNum = int(str(abs(x))[::-1])
```

Comment:
While this solution works, the purpose of this problem is definitly not only this.
This method has a O(log(n)) space complexity (not much since you can't go beyond 2^31). It can be done with O(1) complexity.
Please rework on this problem and provide an alternative solution.

Reply:
done :0